internal/pkg/handlers/containerService.go

```diff
@@ -0,0 +1,330 @@
+package handlers
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/go-openapi/loads"
+
+	"github.com/lawrencegripper/azbrowse/pkg/armclient"
+	"github.com/lawrencegripper/azbrowse/pkg/swagger"
+)
+
+type clusterCredentialsResponse struct {
+	KubeConfigs []struct {
+		Name  string `json:"name"`
+		Value string `json:"Value"`
+	} `json:"kubeconfigs"`
+}
+
+// kubeConfigResponse is a minimal struct for parsing the parts of the response that we care about
+type kubeConfigResponse struct {
+	Clusters []struct {
+		Name    string `yaml:"name"`
+		Cluster struct {
+			CertificateAuthorityData string `yaml:"certificate-authority-data"`
+			Server                   string `yaml:"server"`
+		} `yaml:"cluster"`
+	} `yaml:"clusters"`
+	Users []struct {
+		Name string `yaml:"name"`
+		User struct {
+			ClientCertificateData string `yaml:"client-certificate-data"`
+			ClientKeyData         string `yaml:"client-key-data"`
+			Token                 string `yaml:"token"`
+		} `yaml:"user"`
+	} `yaml:"users"`
+}
+
+// AzureKubernetesServiceExpander expands the kubernetes aspects of AKS
+type AzureKubernetesServiceExpander struct {
+}
+
+// Name returns the name of the expander
+func (e *AzureKubernetesServiceExpander) Name() string {
+	return "AzureKubernetesServiceExpander"
+}
+
+// DoesExpand checks if this is a storage account
+func (e *AzureKubernetesServiceExpander) DoesExpand(ctx context.Context, currentItem *TreeNode) (bool, error) {
+	swaggerResourceType := currentItem.SwaggerResourceType
+	if currentItem.ItemType == "resource" && swaggerResourceType != nil {
+		if swaggerResourceType.Endpoint.TemplateURL == "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerService/managedClusters/{resourceName}" {
+			return true, nil
+		}
+	}
+	if currentItem.Namespace == "AzureKubernetesService" {
+		return true, nil
+	}
+	return false, nil
+}
+
+// Expand returns ManagementPolicies in the StorageAccount
+func (e *AzureKubernetesServiceExpander) Expand(ctx context.Context, currentItem *TreeNode) ExpanderResult {
+
+	swaggerResourceType := currentItem.SwaggerResourceType
+	if currentItem.Namespace != "AzureKubernetesService" &&
+		swaggerResourceType != nil &&
+		swaggerResourceType.Endpoint.TemplateURL == "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerService/managedClusters/{resourceName}" {
```

Comment:
nit: maybe pull out into a const

Reply:
Changed my mind... think it'd be less clear as a const

Reply:
#done

Reply:
It is repeated in a couple of places, so I opted for const :-)